apps/autoscheduler/CMakeLists.txt

```diff
@@ -0,0 +1,157 @@
+# weights
+file(GLOB WF baseline.weights)
+get_filename_component(BASENAME_WF ${WF} NAME_WE)
+set(WF_CPP ${BASENAME_WF}.cpp)
+
+add_custom_command(OUTPUT ${WF_CPP}
+                   DEPENDS ${WF} binary2cpp
+                   COMMAND binary2cpp ${BASENAME_WF}_weights < ${WF} > ${WF_CPP}
+                   COMMENT "${WF} -> ${WF_CPP}"
+                   VERBATIM)
+
+# =======================================================
+
+# retrain_cost_model
+add_executable(retrain_cost_model_process
+               ASLog.cpp
+               DefaultCostModel.cpp
+               Weights.cpp
+               retrain_cost_model.cpp
+               ${WF_CPP})
+set_target_properties(retrain_cost_model_process
+                      PROPERTIES OUTPUT_NAME retrain_cost_model)
+
+set(COST_MODEL_GENERATOR_FILES cost_model_generator.cpp)
+halide_generator(cost_model.generator SRCS ${COST_MODEL_GENERATOR_FILES})
+halide_generator(train_cost_model.generator SRCS ${COST_MODEL_GENERATOR_FILES})
+halide_library_from_generator(cost_model
+                              FUNCTION_NAME cost_model
+                              GENERATOR cost_model.generator)
+halide_library_from_generator(train_cost_model
+                              FUNCTION_NAME train_cost_model
+                              GENERATOR train_cost_model.generator)
+if(MSVC AND (CMAKE_SIZEOF_VOID_P EQUAL 4))
+  set_target_properties(retrain_cost_model_process
+                        PROPERTIES LINK_FLAGS "/FORCE:MULTIPLE")
+endif()
+target_include_directories(retrain_cost_model_process
+                           PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../support)
+target_link_libraries(retrain_cost_model_process
+                      PRIVATE cost_model train_cost_model)
+add_dependencies(retrain_cost_model_process cost_model train_cost_model)
```

Comment:
This should be unnecessary; IIUC, `target_link_libraries` implicitly does `add_dependencies`. (I don't think this is causing the Win32 problem we are seeing, but it still shouldn't be necessary, here and a few other places)